src/mongo/base/global_initializer.cpp

```diff
@@ -24,13 +24,4 @@ namespace mongo {
         return theGlobalInitializer;
     }
 
-namespace {
-
-    // Make sure that getGlobalInitializer() is called at least once before main(), and so at least
-    // once in a single-threaded context.  Otherwise, static initialization inside
-    // getGlobalInitializer() won't be thread-safe.
-    Initializer* _theGlobalInitializer = &getGlobalInitializer();
```

Comment:
You definitely can't do this, as it will stop all global initializers from running. Did you check that mongod actually starts up with these changes?